packages/react-router/modules/Prompt.js

```diff
@@ -39,9 +44,28 @@ class Prompt extends React.Component {
     }
   }
 
+  onbeforeunload(e) {
+    var dialogText = "Changes you made may not be saved."
+    e.returnValue = dialogText
+    return dialogText
+  }
+
+  enableUnload(_onbeforeunload) {
+    if(this._subscribed) { return }
+    this._subscribed = _onbeforeunload ? _onbeforeunload : this.onbeforeunload;
+    window.addEventListener(BEFOREUNLOAD, this._subscribed)
```

Comment:
apologies, I assumed that Prompt will aways be used within a Browser. I'm guessing Prompt will also work in Native? 
All the work with Monorepos and Lerna is making sense :)

There are multiple ways to do this, I am going to start working on HOC as I prefer this solution, unless I run into trouble :) Let me know if you think otherwise

- Inheritance
- Composition
- HOC

It seems over the top to create a new Module just for this one feature? Docs, Examples and new code file (even if it is a micro HOC).
You're the boss so I will make a new component :)